src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -1189,12 +1183,31 @@ public object Get(System.Type entityClass, object id)
 		/// <param name="lockMode"></param>
 		/// <returns></returns>
 		public object Get(System.Type clazz, object id, LockMode lockMode)
+		{
+			return Get(clazz.FullName, id, lockMode);
+		}
+
+		/// <summary>
+		/// Return the persistent instance of the given entity name with the given identifier, or null
+		/// if there is no such persistent instance. (If the instance, or a proxy for the instance, is
+		/// already associated with the session, return that instance or proxy.)
+		/// </summary>
+		public object Get(string entityName, object id, LockMode lockMode)
 		{
 			using (BeginProcess())
 			{
-				LoadEvent loadEvent = new LoadEvent(id, clazz.FullName, lockMode, this);
-				FireLoad(loadEvent, LoadEventListener.Get);
-				return loadEvent.Result;
+				LoadEvent loadEvent = new LoadEvent(id, entityName, lockMode, this);
+				bool success = false;
+				try
+				{
+					FireLoad(loadEvent, LoadEventListener.Get);
+					success = true;
+					return loadEvent.Result;
+				}
+				finally
+				{
+					AfterOperation(success);
+				}
```

Comment:
This turns the PR into a non trivial bug fix by the way, causing the connection to be released after each Get provided with a lock mode in default release mode, if calling these Get(s) outside of a transaction. (Not having the connection released is a bug.) I think it should be mentioned in the PR description.

Reply:
Can you update the PR description the way you think is better?

Reply:
I have checked with a test, and in fact the `AfterStatement` logic is already closing the connection, even without this `AfterOperation` call. 

But it causes quite another change to have that `AfterOperation` call: it will release the lock if we are outside of a transaction.
So this change switches a somewhat working call to a non-working call. The lock will be released at the end of the Get.

It can be argued that acquiring a lock outside a transaction is anyway a mistake, especially since many other operations on the session would cause the `AfterOperation` to release the lock.
I think the only appropriate action here would be to throw when a lock is asked for (other than `None`) while outside of a transaction.
But it is a breaking change, and I think it should be the subject of another PR, probably for 6.0.

Here the change should be reverted because it is still a possible breaking change and unrelated to the PR scope. Likely a comment should be added to warn about this pitfall. I thing the lack of `AfterOperation` was done on purpose, but that is a debatable choice and at least a comment should explain why it is not there.
Something like `AfterOperation call is skipped to avoid releasing the lock when outside of a transaction.`.

But then, it is troublesome to have refactored `Get`s without a lock mode to use the `Get` with a lock mode, because they were not behaving the same and cannot behave the same regarding `AfterOperation`, unless accepting the possible breaking change this PR currently introduces.

Reply:
Could you please clarify what it would break if we just skip `AfterOperation` if `lockMode` is null? As it's just silly that `session.Get<T>(id)` and `session.Get<T>(id, lockMode: null)` behaves differently

Reply:
It may still breaks users not using a transaction (at least not one NHibernate is aware of, like using a transaction scope with a non-scope aware transaction factory) for doing a get with an actual lock mode then using more get with null or None lock mode in order to still keep the previously acquired lock of the first get.